src/commands/lint/configureLintingToolsCommand.ts

```diff
@@ -11,29 +11,46 @@ import * as path from 'path';
 import { WorkspaceUtils } from '../../utils/workspaceUtils';
 import { JSON_INDENTATION_SPACES } from '../../utils/constants';
 
-const config = workspace.getConfiguration();
-const eslintPluginLwcGraphAnalyzer =
-    '@salesforce/eslint-plugin-lwc-graph-analyzer';
-const eslintPluginLwcGraphAnalyzerConfig =
-    'mobileOfflineLinting.eslint-plugin-lwc-graph-analyzer';
-const eslintPluginLwcGraphAnalyzerVersion = config.get(
-    eslintPluginLwcGraphAnalyzerConfig
-) as string;
-
-const eslint = 'eslint';
-const eslintConfig = 'mobileOfflineLinting.eslint';
-const eslintVersion = config.get(eslintConfig) as string;
-
 const configureLintingToolsCommand =
     'salesforcedx-vscode-offline-app.configureLintingTools';
-const eslintDependencies = [
-    [eslintPluginLwcGraphAnalyzer, eslintPluginLwcGraphAnalyzerVersion],
-    [eslint, eslintVersion]
-];
 
-const lwcGraphAnalyzerRecommended: string =
-    'plugin:@salesforce/lwc-graph-analyzer/recommended';
-const eslintRecommended = 'eslint:recommended';
+const config = workspace.getConfiguration();
+
+class EslintDependencyConfig {
```

Comment:
In TypeScript, these kinds of data structures are better represented as types (or [type aliases](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-aliases), technically). I actually like their types [cheat sheet](https://www.typescriptlang.org/static/TypeScript%20Types-ae199d69aeecf7d4a2704a528d0fd3f9.png), once your head is around the basic concept. Nice go-to reference for syntax.

Reply:
sure, let me do that

Reply:
read a bit about type/type aliases, I didn't figure out a way to let type to a member as function.  I think class is still better due to we need to have version value to be dynamically set based.   so uses class with a function getVersion() support it.

Reply:
Yeah, it'd be a bit different in the type setup. Instead of having a "lazy load" method to get the version, you'd simply initialize the version with the logic that it contains. Arguably, that's the only reason the `packageConfigPropertyId` exists. So it would instead look something like this:

```typescript
type EslintDependencyConfig = {
    name: string;
    version: string;
    eslintConfigToExtend: string;
};

const eslintDependencies: EslintDependencyConfig[] = [
    {
        name: '@salesforce/eslint-plugin-lwc-mobile',
        eslintConfigToExtend: 'plugin:@salesforce/lwc-mobile/recommended',
        version: config.get('mobileOfflineLinting.eslint-plugin-lwc-mobile')
    },
    // etc.
];
```

But it's not a big deal. It works fine the way you have it too.